nft/src/lib.rs

```diff
@@ -175,433 +331,386 @@ pub mod pallet {
-		}
-
-		#[pallet::weight(T::WeightInfo::set_nft_mint_fee())]
-		pub fn set_nft_mint_fee(
-			origin: OriginFor<T>,
-			mint_fee: BalanceOf<T>,
-		) -> DispatchResultWithPostInfo {
-			ensure_root(origin)?;
-
-			NFTMintFee::<T>::put(mint_fee);
+			let event = Event::NFTTransferred { nft_id, sender: who, recipient };
 
-			Self::deposit_event(Event::NFTMintFeeUpdated { fee: mint_fee });
+			Self::deposit_event(event);
 
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::delegate())]
-		pub fn delegate(
+		/// Delegate an NFT to a recipient, does not change ownership
+		/// Must be called by NFT owner
+		#[pallet::weight(T::WeightInfo::delegate_nft())]
+		pub fn delegate_nft(
 			origin: OriginFor<T>,
 			nft_id: NFTId,
-			viewer: Option<T::AccountId>,
+			recipient: Option<<T::Lookup as StaticLookup>::Source>,
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
+			let mut recipient_account_id: Option<T::AccountId> = None;
 
-			Data::<T>::try_mutate(nft_id, |maybe_data| -> DispatchResult {
-				let data = maybe_data.as_mut().ok_or(Error::<T>::NFTNotFound)?;
+			NFTs::<T>::try_mutate(nft_id, |maybe_nft| -> DispatchResult {
+				let nft = maybe_nft.as_mut().ok_or(Error::<T>::NFTNotFound)?;
 
-				ensure!(data.owner == who, Error::<T>::NotTheNFTOwner);
-				ensure!(!data.listed_for_sale, Error::<T>::CannotDelegateNFTsListedForSale);
-				ensure!(!data.is_capsule, Error::<T>::CannotDelegateCapsules);
-				ensure!(!data.is_in_transmission, Error::<T>::CannotDelegateNFTsInTransmission);
+				// Checks
+				ensure!(nft.owner == who, Error::<T>::NotTheNFTOwner);
+				ensure!(!nft.state.listed_for_sale, Error::<T>::CannotDelegateListedNFTs);
+				ensure!(!nft.state.is_capsule, Error::<T>::CannotDelegateCapsuleNFTs);
 
-				if let Some(viewer) = &viewer {
-					ensure!(who != *viewer, Error::<T>::CannotDelegateNFTsToYourself);
+				if let Some(recipient) = &recipient {
+					let recipient = T::Lookup::lookup(recipient.clone())?;
+					ensure!(who != recipient, Error::<T>::CannotDelegateNFTsToYourself);
+					recipient_account_id = Some(recipient);
 				}
-				let data = maybe_data.as_mut().ok_or(Error::<T>::NFTNotFound)?;
-				data.is_delegated = viewer.is_some();
+
+				// Execute
+				nft.state.is_delegated = recipient_account_id.is_some();
 
 				Ok(().into())
 			})?;
 
-			match viewer.as_ref() {
+			// Execute
+			match recipient_account_id.as_ref() {
 				Some(v) => DelegatedNFTs::<T>::insert(nft_id, v),
 				None => DelegatedNFTs::<T>::remove(nft_id),
 			}
 
-			let event = Event::NFTDelegated { nft_id, viewer };
+			let event = Event::NFTDelegated { nft_id, recipient: recipient_account_id };
 			Self::deposit_event(event);
 
 			Ok(().into())
 		}
-	}
 
-	#[pallet::event]
-	#[pallet::generate_deposit(pub(super) fn deposit_event)]
-	pub enum Event<T: Config> {
-		/// A new NFT was created.
-		NFTCreated {
+		/// Set the royalty of an NFT
+		/// Can only be called from creator if creator is the owner.
```

Comment:
Rephrase. Can only be called if the NFT is owned and has been created by the caller.